### PR TITLE
optionally read rocksdb configfile

### LINKED ISSUE #18: Configure database through RocksDB options file
Can we initialize the fossildb through the RocksDB options file?
The benefit would be that we can customize all RocksDB options, including compaction parameters.
To create new databases, we could ship an options file with the project.

See https://github.com/facebook/rocksdb/blob/master/java/src/main/java/org/rocksdb/OptionsUtil.java#L89

### PR DESCRIPTION
**Pull request by fm3**
new command line option `-r <filepath>` causes a [rocksdb options file](https://github.com/facebook/rocksdb/wiki/RocksDB-Options-File#rocksdb-options-file-format) to be read.

The two options `create_if_missing` and `create_missing_column_families` from the file are ignored and are always set to `true`.

We still use (and create) the column families specified by the command line parameter `-c` (rather than constructing them from the config file. We will change that in a future PR)

A sample rocks-config.ini looks like this
```
[Version]
  rocksdb_version=5.11.3
  options_file_version=1.1

[DBOptions]
  stats_dump_period_sec=700

[CFOptions "default"]

```

fixes #18

### CODE REVIEW

File: src/main/scala/com/scalableminds/fossildb/db/RocksDBStore.scala [hunk 1]
```diff
@@ -28,9 +29,21 @@ class RocksDBManager(dataDir: Path, columnFamilies: List[String]) extends LazyLo
       new ColumnFamilyDescriptor(columnFamily, columnOptions)
     }
     val columnFamilyHandles = new util.ArrayList[ColumnFamilyHandle]
-    val options = new DBOptions()
+    var options = new DBOptions()
       .setCreateIfMissing(true)
       .setCreateMissingColumnFamilies(true)
+    var cfListRef: mutable.Buffer[ColumnFamilyDescriptor] = mutable.Buffer()
```
Comment (jstriebel) [inline src/main/scala/com/scalableminds/fossildb/db/RocksDBStore.scala]: @fm3: this is not used currently, right?

Comment (fm3) [reply src/main/scala/com/scalableminds/fossildb/db/RocksDBStore.scala]: correct, but it has to exist and to be passed (by reference) to the loadOptionsFromFile

**Comment by normanrz** [comment]
What is the reason for keeping `-c`?

**Comment by fm3** [comment]
@jstriebel I added some tests

**Pull request merged**